src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -0,0 +1,77 @@
+package ru.hogwarts.school.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ru.hogwarts.school.model.Student;
+
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.service.implemention.FacultyServiceImpl;
+
+import java.util.Collection;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("faculty")
+public class FacultyController {
+
+
+    private final FacultyServiceImpl facultyService;
+
+@Autowired
+    public FacultyController (FacultyServiceImpl facultyService) {
+        this.facultyService = facultyService;
+    }
+
+
+    @GetMapping("{id}")
+    public ResponseEntity<Optional<Faculty>> getFacultyById(@PathVariable Long id) {
+        Optional<Faculty> faculty = facultyService.findFaculty(id);
+        if (faculty == null) {
```

Comment:
```suggestion
        if (faculty.isEmpty()) {
```
Нужно условие поменять